src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -34,13 +34,13 @@ namespace Microsoft.IdentityModel.Tokens
     /// Creates <see cref="SignatureProvider"/>s by specifying a <see cref="SecurityKey"/> and algorithm.
     /// <para>Supports both <see cref="AsymmetricSecurityKey"/> and <see cref="SymmetricSecurityKey"/>.</para>
     /// </summary>
-    public class SignatureProviderFactory
+    public class CryptoProviderFactory
```

Comment:
> Also both of these providers share the basic attributes - key and algorithm, so it makes sense to have a common wrapper for them.

Why not. But semantically, it seems really odd, as signing and encrypting routines are really different (algorithms too, BTW. For instance, there's no encrypting equivalent for ECDsa).